### ISSUE TITLE
how to include `~/.bashrc` when launching warp

### ISSUE THREAD

**Issue by vchong** [warpdotdev/Warp#5540]
### Dupe Check

- [X] I have [searched Warp feature requests](https://github.com/warpdotdev/warp/issues?q=is%3Aissue+label%3A%22Feature+requests%22) and there are no duplicates

### Describe the solution you'd like?

i'd like to have all the customization (aliases, functions, etc) in `~/.bashrc` available when warp is launched

### How important is this feature to you?

5 (Can't work without it!)

### Warp Internal (ignore) - linear-label:39cc6478-1249-4ee7-950b-c428edfeecd1

None

**Comment by dannyneira** [comment]
Hey @vchong thanks for submitting. Warp supports loading the rc files on launch. Just make sure you have bash as your default prompt in `Settings > Features > Startup shell for new sessions`. https://docs.warp.dev/getting-started/using-warp-with-shells#changing-what-shell-warp-uses

Also, if you have a custom shell prompt setup in your rc files, go to `Settings > Appearance > Prompt` and select your Shell prompt (PS1). https://docs.warp.dev/appearance/prompt#how-to-access-it

Once you're done with those settings just open a new session and the changes should apply.

Closing as this feature exists, but please let me know if any issues with it.

**Comment by dannyneira** [comment]
Hm this may be due to multiple versions of bash on your system, run `which -a bash` and put the top result into the `Startup shell for new sessions > Custom` setting. Also if you can share a copy of your bashrc we can try and reproduce the issue. Make sure to not include any important info like secrets or API keys.

**Comment by dannyneira** [comment]
Yeah sorry about that, i suspect you may have something else unsupported in your bashrc file causing the issue. Can you please share a copy of your .bashrc so that we can try and reproduce the issue? Make sure to not include any important info like secrets or API keys.

We have a list of known incompatible tools here: <https://docs.warp.dev/help/known-issues#list-of-incompatible-tools>

**Comment by dannyneira** [comment]
Thanks for the info @vchong I'm having trouble reproducing this on the latest version of bash on Mac. 

I recommend you install the latest version of bash (i.e. `sudo apt upgrade ; sudo apt install bash` on Ubuntu for example), then make sure that version is being used in your settings. If this fails please [provide the logs](https://docs.warp.dev/help/sending-us-feedback#gathering-warp-logs) for this issue.

![CleanShot 2024-12-09 at 12 44 42](https://github.com/user-attachments/assets/adb5d61d-4d81-4bd1-b012-820140398a97)

**Comment by dannyneira** [comment]
@vchong i see, I tested both on Ubuntu 24 with bash 5.2 and Ubuntu 22 with bash 5.1 and both didn't have the same issue as you. I suspect there may be something in your ~/.bash_profile, which is loaded before the ~/.bashrc may be causing this issue. If not, then it could be the settings in Warp. Also, in the past, there were issues with the `--color=auto` option, but as far as I know these issues have been fixed.

First, check your `~/.bash_profile` for anything in the list of incompatible tools. https://docs.warp.dev/help/known-issues#list-of-incompatible-tools
If any, you can comment it out for Warp with the following snippet:
```sh
if [[ $TERM_PROGRAM != "WarpTerminal" ]]; then
    # Unsupported plugin/prompt code here, i.e.
fi
```

If this doesn't help, I recommend you try a full uninstall, including any settings, database, and files, then reinstall and see if the issue persists. https://docs.warp.dev/help/uninstalling-warp#uninstalling-warp

**Comment by dannyneira** [comment]
@vchong any luck searching thru your bashrc file?

**Comment by turbotimon** [comment]
My system didn't hat a `~/.bash_profile` yet, only `~/.bashrc` and `~/.bash_aliases`.

Creating a bash_profile helped:

```sh
# ~/.bash_profile
if [ -f ~/.bashrc ]; then
    source ~/.bashrc
fi
```

However, I think warp should handle this...
